deploy/utils/setup_k8s_namespace.sh

```
@@ -0,0 +1,195 @@
+  ok "hf-token-secret created/updated"
+fi
+
+# 5) Optional: Create imagePullSecret for private registry if credentials provided or requested
+if [[ -n "$DOCKER_SERVER" ]]; then
+  if [[ -n "$DOCKER_USERNAME" && -n "$DOCKER_PASSWORD" ]]; then
+    create_or_update_pull_secret "$DOCKER_SERVER" "$DOCKER_USERNAME" "$DOCKER_PASSWORD"
+  elif [[ -n "$IMAGE_TAG" ]]; then
+    echo
+    read -p "Do you need image pull credentials for $DOCKER_SERVER (private registry)? [y/N]: " -r ans
+    if [[ "$ans" =~ ^[Yy]$ ]]; then
+      read -p "Docker username (often '$oauthtoken' for NGC): " DOCKER_USERNAME
```

Shellcheck (0.10.0)

[warning] oauthtoken is referenced but not assigned.

(SC2154)